.flake8

```diff
@@ -0,0 +1,4 @@
+[flake8]
+max-line-length = 120
+ignore = D100, D103, D104, D401, E203, E501, E731, E741, F401, W503
```

Comment:
What do these ignore flags mean?

Reply:
A list can be found [here](https://martin-thoma.com/python-style-guide/). I'll summarize the ones ignored:

Docstrings:
* D100: Missing docstring in public module
* D103: Missing docstring in public function
* D104: Missing docstring in public package

Basically, I'm enforcing that public classes and methods have docstrings, but not enforcing that anything else has them or it may become cumbersome. Upon reflection, it actually makes sense to also enforce public functions to have docstrings if we have a bunch of utils, so I'll probably remove D103 from the ignore list.

Formatting:
* E203: Whitespace before `:`
* E501: Line too long (79) (ignore the lower line length limit and set a higher one, 120)
* E731: Do not assign a lambda expression
* E741: Do not use variables named `l`, `O`, or `I`
* W503: Line break before binary operator (when breaking `a+b` into multiple lines, leave the `+` on top line)

`flake8`
* F401: module imported but unused

Reply:
Update: removed D103 and also allow unused imports in the `autoflake` config in the pre-commit yaml file